src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonSchemaCreateOptions.cs

```diff
@@ -27,4 +27,25 @@ public sealed class AIJsonSchemaCreateOptions
     /// Gets a value indicating whether to include the $schema keyword in inferred schemas.
     /// </summary>
     public bool IncludeSchemaKeyword { get; init; }
+
+    /// <summary>
+    /// Gets a value indicating whether to mark all properties as required in the schema.
+    /// </summary>
+    public bool RequireAllProperties { get; init; }
+
+    /// <summary>
+    /// Gets a value indicating whether to filter keywords that are disallowed by certain AI vendors.
+    /// </summary>
+    /// <remarks>
+    /// Filters a number of non-essential schema keywords that are not yet supported by some AI vendors.
+    /// These include:
+    /// <list type="bullet">
+    /// <item>The "minLength", "maxLength", "pattern", and "format" keywords.</item>
+    /// <item>The "minimum", "maximum", and "multipleOf" keywords.</item>
+    /// <item>The "patternProperties", "unevaluatedProperties", "propertyNames", "minProperties", and "maxProperties" keywords.</item>
+    /// <item>The "unevaluatedItems", "contains", "minContains", "maxContains", "minItems", "maxItems", and "uniqueItems" keywords.</item>
+    /// </list>
+    /// See also https://platform.openai.com/docs/guides/structured-outputs#some-type-specific-keywords-are-not-yet-supported.
+    /// </remarks>
+    public bool FilterDisallowedKeywords { get; init; }
```

Comment:
Should this instead be an enumerable / list of keywords to filter? It feels a bit skewed that we're hardcoding which keywords are disallowed.

Reply:
I tried that initially, but putting any sort of collection in the options messes up with our schema caching strategy which keys on the configuration settings. I could try to do that, but it would require building a custom list with sequence equality semantics.

Reply:
I see. I'm concerned that this will quickly get stale. Compat-wise, this will effectively always need to represent the same set of keywords, even as constraints evolve.